arch/xtensa/src/esp32s2/esp32s2_oneshot.h

```diff
@@ -0,0 +1,204 @@
+ * Public Types
+ ****************************************************************************/
+
+/* This describes the callback function that will be invoked when the oneshot
+ * timer expires.  The oneshot fires, the client will receive:
+ *
+ *   arg - The opaque argument provided when the interrupt was registered
+ */
+
+typedef void (*oneshot_handler_t)(void *arg);
+
+/* The oneshot client must allocate an instance of this structure and call
+ * esp32s2_oneshot_initialize() before using the oneshot facilities.  The
+ * client should not access the contents of this structure directly since
+ * the contents are subject to change.
+ */
+
+struct esp32s2_oneshot_s
+{
+  uint8_t chan;                         /* The timer/counter in use */
+  volatile bool running;                /* True: the timer is running */
+  struct esp32s2_tim_dev_s        *tim; /* Pointer returned by
+                                         * esp32s2_tim_init() */
+  volatile oneshot_handler_t   handler; /* Oneshot expiration callback */
+  volatile void                   *arg; /* The argument that will accompany
+                                         * the callback */
+  uint32_t                  resolution; /* us */
+};
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32s2_oneshot_initialize
+ *
+ * Description:
+ *   Initialize the oneshot timer wrapper.
+ *
+ * Input Parameters:
+ *   oneshot    Allocated instance of the oneshot state structure.
+ *   chan       Timer counter channel to be used.
+ *   resolution The required resolution of the timer in units of
+ *              microseconds.  NOTE that the range is restricted to the
+ *              range of uint16_t (excluding zero).
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on failure.
+ *
+ ****************************************************************************/
+
+int esp32s2_oneshot_initialize(struct esp32s2_oneshot_s *oneshot,
+                               int chan, uint16_t resolution);
+
+/****************************************************************************
+ * Name: esp32s2_oneshot_max_delay
+ *
+ * Description:
+ *   Determine the maximum delay of the one-shot timer (in microseconds).
+ *
+ * Input Parameters:
+ *   oneshot    Allocated instance of the oneshot state structure.
+ *   chan       The location in which to return the maximum delay in us.
```

Comment:
```suggestion
 *
 * Output Parameters:
 *   usec    The maximum delay in us.
```
Incorrect mention to `chan` param. And `usec` is an output parameter here.